modules/installation-osp-local-disk-deployment.adoc

```diff
@@ -0,0 +1,296 @@
+// Module included in the following assemblies:
+//
+// * installing_openstack/deploying-openstack-with-rootVolume-etcd-on-local-disk.adoc
+
+:_mod-docs-content-type: Procedure
+[id="installation-osp-local-disk-deployment_{context}"]
+= Deploying {rh-openstack} on local disk
+
+.Prerequisites
+
+* You have an OpenStack cloud with a working Cinder, and it has at least 75 GB of available storage to accommodate 3 root volumes for the OpenShift control plane.
+
+* The OpenStack cloud must be deployed with Nova ephemeral storage that uses a local storage backend and not RBD.
+
+
+.Procedure
+
+[WARNING]
+====
+This procedure can testing etcd on local disk and should not be used on production cluster because in certain cases complete loss of the control plane can occur. For more information, see "Overview of backup and restore operation" under "Backup and restore".
+====
+
+. Create a Nova flavor for the control plane with 10 GB of ephemeral disk by running the following command:
++
+[source,terminal]
+----
+$ openstack flavor create --ram 16384 --disk 25 --ephemeral 10 --vcpus 4 m1.xlarge
+----
+
+. Deploy a cluster with root volumes for the control plane; for example:
++
+.Example YAML file
+[source,yaml]
+----
+# ...
+controlPlane:
+  name: master
+  platform:
+    openstack:
+      type: ${CONTROL_PLANE_FLAVOR}
+      rootVolume:
+        size: 25
+        types:
+        - ${CINDER_TYPE}
+  replicas: 3
+# ...
+----
+
+. Deploy the cluster you created by running the following command:
++
+[source,terminal]
+----
+$ openshift-install create cluster --dir <installation_directory> <1>
+----
++
+<1> For `<installation_directory>`, specify the location of your customized `./install-config.yaml file` that you created in step 2.
++
+[NOTE]
+====
+Ensure that the cluster you deployed is healthy before proceeding to step 4.
+====
+
+. Edit the `ControlPlaneMachineSet` (CPMS) to add the additional block ephemeral device that will be used by etcd by running the following command:
++
+[%collapsible]
+====
+[source,terminal]
+----
+$ oc patch ControlPlaneMachineSet/cluster -n openshift-machine-api --type json -p ' <1>
+[
+    {
+      "op": "add",
+      "path": "/spec/template/machines_v1beta1_machine_openshift_io/spec/providerSpec/value/additionalBlockDevices", <2>
+      "value": [
+        {
+          "name": "etcd",
+          "sizeGiB": 10,
+          "storage": {
+            "type": "Local" <3>
+          }
+        }
+      ]
+    }
+  ]
+'
+----
+<1> Applies the JSON patch to the `ControlPlaneMachineSet` CR.
+<2> Specifies the path where the `additionalBlockDevices` are added.
+<3> Adds the `etcd` devices with local storage of 10 GB to the cluster.
```

Comment:
with at least 10 GB. Note that the size of it must fit from the Nova flavor.
If the flavor has 15 GB, you can create the device with 12 GB.
But if the flavor has 10 GB, the maximum for etcd device will be 10 GB.

Reply:
I'd also be interested to test what happens if etcd grows beyond this size. How could we safely recover the cluster in this case? It could involve some interesting bootstrapping problems.

We should test and document this for GA, but I don't think it's a TP blocker.

Reply:
I like Matt's point. We should capture it in this doc, we'll add that in our GA EPIC (when we have one...).
Maybe add a warning that the procedure to make etcd's storage grow above the ephemeral disk size isn't figured out yet and will be when the feature is GA.